src/widgets/factory/service_row.rs

```diff
@@ -0,0 +1,145 @@
+use crate::application::plugin::Plugin;
+use crate::fl;
+use crate::widgets::components::preferences::{
+	PluginPreferences, PreferencesComponentInput,
+};
+use adw::prelude::{ActionRowExt, BoxExt, ButtonExt, PreferencesRowExt};
+use relm4::adw;
+use relm4::factory::AsyncFactoryComponent;
+use relm4::factory::{AsyncFactorySender, DynamicIndex, FactoryView};
+use relm4::gtk;
+use relm4::gtk::traits::WidgetExt;
+
+#[derive(Debug, Default)]
+pub struct ServiceRowModel {
+	pub plugin: Plugin,
+	pub enabled: bool,
+	pub installed: bool,
+	pub first_load: bool,
+}
+
+#[derive(Debug)]
+pub enum ServiceRowInput {
+	InstallPlugin(DynamicIndex),
+	EnablePlugin(DynamicIndex),
+	DisablePlugin(DynamicIndex),
+	HideInstallButton,
+	EnableSwitch(bool),
+}
+
+#[derive(Debug)]
+pub enum ServiceRowOutput {
+	InstallPlugin(DynamicIndex, Plugin),
+	EnablePlugin(DynamicIndex, Plugin),
+	DisablePlugin(DynamicIndex, Plugin),
+}
+
+#[relm4::factory(pub async)]
+impl AsyncFactoryComponent for ServiceRowModel {
+	type ParentInput = PreferencesComponentInput;
+	type ParentWidget = adw::PreferencesGroup;
+	type CommandOutput = ();
+	type Input = ServiceRowInput;
+	type Output = ServiceRowOutput;
+	type Init = PluginPreferences;
+	type Widgets = ProviderWidgets;
+
+	view! {
+				#[root]
+		#[name(service)]
+				adw::ActionRow {
```

Comment:
Another real-world use-case for relm4-format :P

Reply:
Lol. fixed!